parcels/particleset.py

```diff
@@ -148,7 +149,10 @@ def ArrayClass_init(self, *args, **kwargs):
             pid_orig = np.arange(lon.size)
 
         if depth is None:
-            mindepth = self.fieldset.gridset.dimrange("depth")[0]
+            if type(self.fieldset) == UXFieldSet:
+                mindepth = 0  # TO DO :  get the min depth from the fieldset.uxgrid
```

Comment:
I suppose we'll need a new implementation for `fieldset.gridset.dimrange` . Since the new proposed fieldset is meant to wrap around either an XArray or UXArray dataset, which each have a grid baked into the structure, the `gridset` subclass here is irrelevant.